index.js

```diff
@@ -63,16 +65,15 @@ let customRandom = (alphabet, size, getRandom) => {
 let customAlphabet = (alphabet, size) => customRandom(alphabet, size, random)
 
 let nanoid = (size = 21) => {
-  let bytes = random(size)
+  _fillPool(size)
   let id = ''
-  // A compact alternative for `for (let i = 0; i < size; i++)`.
-  while (size--) {
+  for (let i = poolOffset - size; i < poolOffset; i++) {
```

Comment:
We do not update `poolOffset` after using the random value.

Will it lead to reusing values from the pull between `nanoid` calls?